app/waf/test/acceptance/service.controller.acceptance.ts

```diff
@@ -1,44 +1,87 @@
 import {Client, expect, toJSON} from '@loopback/testlab';
 import {WafApplication} from '../..';
-import {setupApplication, teardownApplication} from '../helpers/test-helper';
+import {
+  setupApplication,
+  teardownApplication,
+  TestingApplication,
+  setupRestAppAndClient,
+  RestApplicationPort,
+  teardownRestAppAndClient,
+} from '../helpers/test-helper';
 import {
   givenEmptyDatabase,
   givenServiceData,
 } from '../helpers/database.helpers';
+import {
+  ShouldResponseWith,
+  MockKeyStoneController,
+  ExpectedData,
+} from '../fixtures/controllers/mocks/mock.openstack.controller';
 import uuid = require('uuid');
 
 describe('ServiceController', () => {
   let wafapp: WafApplication;
   let client: Client;
+  let mockKeystoneApp: TestingApplication;
 
   const prefix = '/adcaas/v1';
 
+  let envs: {[key: string]: string} = {
+    OS_AUTH_URL: 'http://localhost:35357/v2.0',
+    OS_USERNAME: 'wafaas',
+    OS_PASSWORD: '91153c85b8dd4147',
+    OS_TENANT_ID: '32b8bef6100e4cb0a984a7c1f9027802',
+    OS_DOMAIN_NAME: 'Default',
+    OS_REGION_NAME: 'RegionOne',
+    OS_AVAILABLE_ZONE: 'nova',
+  };
+
   before('setupApplication', async () => {
+    mockKeystoneApp = await (async () => {
+      let {restApp} = await setupRestAppAndClient(
+        RestApplicationPort.IdentityAdmin,
+        MockKeyStoneController,
+      );
+      return restApp;
+    })();
+
     ({wafapp, client} = await setupApplication());
+    ShouldResponseWith({});
+    setupEnvs();
   });
+
   beforeEach('Empty database', async () => {
     await givenEmptyDatabase(wafapp);
   });
 
-  after(async () => {
-    await teardownApplication(wafapp);
-  });
+  let setupEnvs = async () => {
```

Comment:
-> `test-helper.ts`?

Reply:
Yes, it should be in test-helper.ts, this will affect other tests. Do you think it's better to do it after mutli-tenancy applied to all APIs?

Reply:
OK for that. 
It doesn't affect others, I think. 
Only moving the definition to test-helper.ts, while the call to them is still in *.controller.acceptance.ts.

Reply:
I got your idea. You mean to move the definition to the test-helper.ts. I can change all of these in the next patch.